### DISCUSSION TITLE
Added BCM94352Z (DW1560).  Can Apple Watch Unlock work with this or BCM94360CS2 card?

### DISCUSSION THREAD

**Discussion by MarkMorrow** [simprecicchiani/ThinkPad-T460s-macOS-OpenCore#80] [Q&A] [answered]
My T460s already had a DW1560 card installed and working with Clover booting High Sierra / Win10.  I followed this guide and converted to OC booting Big Sur (11.1) / Win10.  It works so much better, boots quicker and feels more responsive.  I'll be sending some coffee dollars to Simprecicchiani in thanks.  :)

As I expected, Big Sur only had ethernet working ( no WiFi or BT).  Adding four kext files resolved that issue, 
**BUT still can't unlock Big Sur with an Apple watch.  Is that possible with the DW1560 card or with the recommended BCM94360CS2 card?**

I've ordered the later to try, but would prefer to stick with the DW1560 if it makes no difference, for it does not require the adapter, and is already in place.

Here is what was added within EFI\OC\kexts for the DW1560 in case others are in the same situation:
[AirportBrcmFixup](https://github.com/acidanthera/AirportBrcmFixup/releases) release v2.1.2 for: **AirportBrcmFixup.kext**, and
[BrcmPatchRAM](https://github.com/acidanthera/BrcmPatchRAM/releases) release v2.5.5 for these three kexts: **BrcmBluetoothInjector.kext**, **BrcmFirmwareData.kext** and **BrcmPatchRAM3.kext**

You then need to add these entries to the config.plist file under the Root->Kernel->Add section.  Alternatively you can use OpenCore Configurator, and drag the kexts to the "add" secition's grid off of the kernel section.  All need to be set to enabled, except for the AirPortBrcm4360_Injector plug-in.

```
	<dict>
		<key>Arch</key>
		<string>Any</string>
		<key>BundlePath</key>
		<string>AirportBrcmFixup.kext</string>
		<key>Comment</key>
		<string>DW1560 WiFi driver</string>
		<key>Enabled</key>
		<true/>
		<key>ExecutablePath</key>
		<string>Contents/MacOS/AirportBrcmFixup</string>
		<key>MaxKernel</key>
		<string></string>
		<key>MinKernel</key>
		<string></string>
		<key>PlistPath</key>
		<string>Contents/Info.plist</string>
	</dict>
	<dict>
		<key>Arch</key>
		<string>Any</string>
		<key>BundlePath</key>
		<string>AirportBrcmFixup.kext/Contents/PlugIns/AirPortBrcm4360_Injector.kext</string>
		<key>Comment</key>
		<string>DW1560 WiFi 4360 plug-in</string>
		<key>Enabled</key>
		<false/>
		<key>ExecutablePath</key>
		<string></string>
		<key>MaxKernel</key>
		<string></string>
		<key>MinKernel</key>
		<string></string>
		<key>PlistPath</key>
		<string>Contents/Info.plist</string>
	</dict>
	<dict>
		<key>Arch</key>
		<string>Any</string>
		<key>BundlePath</key>
		<string>AirportBrcmFixup.kext/Contents/PlugIns/AirPortBrcmNIC_Injector.kext</string>
		<key>Comment</key>
		<string>DW1560 WiFi NIC plug-in</string>
		<key>Enabled</key>
		<true/>
		<key>ExecutablePath</key>
		<string></string>
		<key>MaxKernel</key>
		<string></string>
		<key>MinKernel</key>
		<string></string>
		<key>PlistPath</key>
		<string>Contents/Info.plist</string>
	</dict>
	<dict>
		<key>Arch</key>
		<string>x86_64</string>
		<key>BundlePath</key>
		<string>BrcmBluetoothInjector.kext</string>
		<key>Comment</key>
		<string>DW1560 BT 1 of 3</string>
		<key>Enabled</key>
		<true/>
		<key>ExecutablePath</key>
		<string></string>
		<key>MaxKernel</key>
		<string></string>
		<key>MinKernel</key>
		<string></string>
		<key>PlistPath</key>
		<string>Contents/Info.plist</string>
	</dict>
	<dict>
		<key>Arch</key>
		<string>x86_64</string>
		<key>BundlePath</key>
		<string>BrcmFirmwareData.kext</string>
		<key>Comment</key>
		<string>DW1560 BT 2 of 3</string>
		<key>Enabled</key>
		<true/>
		<key>ExecutablePath</key>
		<string>Contents/MacOS/BrcmFirmwareData</string>
		<key>MaxKernel</key>
		<string></string>
		<key>MinKernel</key>
		<string></string>
		<key>PlistPath</key>
		<string>Contents/Info.plist</string>
	</dict>
	<dict>
		<key>Arch</key>
		<string>x86_64</string>
		<key>BundlePath</key>
		<string>BrcmPatchRAM3.kext</string>
		<key>Comment</key>
		<string>DW1560 BT 3 of 3</string>
		<key>Enabled</key>
		<true/>
		<key>ExecutablePath</key>
		<string>Contents/MacOS/BrcmPatchRAM3</string>
		<key>MaxKernel</key>
		<string></string>
		<key>MinKernel</key>
		<string></string>
		<key>PlistPath</key>
		<string>Contents/Info.plist</string>
	</dict>
```

![image](https://user-images.githubusercontent.com/6678604/102928478-c22cc900-4466-11eb-8e0c-df3f2668ddb8.png)

**Comment by MarkMorrow** [comment]
Some additional research found reports that a BCM94360NG; a M.2 form factor BT4.0, direct replacement card will work with watch unlock, at least on Catalina.  I'm reluctant to pick one up to try for they are $45-50.  Perhaps someone with one and Big Sur could confirm this as a solution, or not.

I'm still waiting on a BCM94360CS2 (lost in the mail?), but based on similar number I suspect this is same chip just needing an adapter as opposed to "NG" that does not?  Perhaps I can answer my own question by just installing and trying once that arrives.

**Comment by simprecicchiani** [comment]
Hi, thank you so much for the donation and the extensive guide. I've never had the occasion to try Apple Watch unlock feature but I guess it will be fine with a genuine BCM94360CS2 card.
Regarding your second question, it looks like it is indeed the same chip on a different form factor.

**Comment by simprecicchiani** [comment, marked as answer]
As #60, Apple Watch unlock feature seems to work even with a DW1560 card.

**Comment by MarkMorrow** [comment]
Watch unlock "works" with a DW1560 if tested right after the option is checked, BUT then it stop working over time and that needs to be repeated to get it working again.  Not sure if it is sleep or something else breaks it, though the feature is not that useful as it is.  Perhaps there is something still needed if this is not the behavior others are seeing.

Once I can find a BCM94360NG card for a reasonable price I will try switching to that and report back.
